File: src/lib/auditoria.ts
import { prisma } from "@/lib/prisma";

export async function registrarAuditoria(
  usuarioId: number | null | undefined,
  email: string | null | undefined,
  accion: string,
  modulo: string,
  detalle?: string
) {
  try {
    await prisma.auditoriaLog.create({
      data: {
        usuarioId: usuarioId ?? undefined,
        email: email ?? undefined,
        accion,
        modulo,
        detalle,
      },
    });
  } catch {
    // no bloquear operación principal
  }
}

File: backend/core/services/auditoria.py
from django.utils import timezone

from core.models import AuditoriaLog


def registrar_auditoria(usuario_id, email, accion, modulo, detalle=""):
    AuditoriaLog.objects.create(
        usuarioId=usuario_id,
        email=email,
        accion=accion,
        modulo=modulo,
        detalle=detalle,
        createdAt=timezone.now(),
    )
